src/test/suite/extension.test.ts

```diff
@@ -1,12 +1,18 @@
 import * as assert from 'assert';
+import * as proxyquire from 'proxyquire';
+import { MockingUtils, MockingExec } from './MockingUtils';
 
 // You can import and use all API from the 'vscode' module
 // as well as import your extension to test it
-import * as vscode from 'vscode';
+//import * as vscode from 'vscode';
+const vscode = require('vscode');
+const mockedExec = new MockingExec();
+proxyquire('child_process', {
```

Comment:
Shouldn't we use `proxyquire` to load `DafnyInstaller` ?
```javascript
const DafnyInstaller = proxyquire('DafnyInstaller', { 'child_process': { exec: mockedExec.stub } }
```

The current code makes it look like `child_process` is mutated for the remainder of the testing process.

Reply:
Yes, thanks for your suggestion, I did not realize I could do this

However, my original code is intended. I don't want the code to actually run the original exec in any portion, so that I can test all exec commands. Hence, I mock the exec for the remaining of the file.
Every test can specify the behavior of exec that it expects.

Reply:
> Hence, I mock the exec for the remaining of the file.

What makes you think this is just the remainder of the file? Wouldn't it mock exec for the entire test process including other files?

Reply:
I return the question to you. What makes you think it would mock exec for the entire test process? Proxyquire is local.

Reply:
Ok, I found a way to make it import locally after running the document. I was probably wrong.